compiler/aarch64/codegen/UnaryEvaluator.cpp

```diff
@@ -54,10 +54,42 @@ TR::Register *OMR::ARM64::TreeEvaluator::cconstEvaluator(TR::Node *node, TR::Cod
    }
 
 TR::Register *OMR::ARM64::TreeEvaluator::aconstEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::aconstEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
+   TR::Compilation *comp = cg->comp();
+   bool isClass = node->isClassPointerConstant();
+   TR_ResolvedMethod * method = comp->getCurrentMethod();
+
+   bool isPicSite = node->isClassPointerConstant() && cg->fe()->isUnloadAssumptionRequired((TR_OpaqueClassBlock *) node->getAddress(), method );
+   if (!isPicSite)
+      isPicSite = node->isMethodPointerConstant() && cg->fe()->isUnloadAssumptionRequired(cg->fe()->createResolvedMethod(cg->trMemory(), (TR_OpaqueMethodBlock *) node->getAddress(), method)->classOfMethod(), method);
+
+   bool isProfiledPointerConstant = node->isClassPointerConstant() || node->isMethodPointerConstant();
+
+   // use data snippet only on class pointers when HCR is enabled
+   intptrj_t address = node->getLongInt();
+   if (isClass && cg->wantToPatchClassPointer((TR_OpaqueClassBlock*)address, node) ||
+       isProfiledPointerConstant && cg->profiledPointersRequireRelocation())
+      {
+      TR::Register *trgReg = cg->allocateRegister();
+      TR_ASSERT(false, "Not implemented yet.");
+      // loadAddressConstantInSnippet();
+      node->setRegister(trgReg);
+      return trgReg;
+      }
+   else
+      {
+      TR::Register *tempReg = node->setRegister(cg->allocateRegister());
+      TR::Instruction *cursor = loadConstant64(cg, node, address, tempReg);
+      if (isPicSite)
+         {
+         if (node->getOpCode().isCall() || node->isClassPointerConstant())
+            comp->getStaticPICSites()->push_front(cursor);
+         else if (node->isMethodPointerConstant())
+            comp->getStaticMethodPICSites()->push_front(cursor);
+         }
+      return tempReg;
```

Comment:
Same comment as above.  This is potentially pulling in a lot of technical debt into AArch64.  Are you able to confirm that all of this will work this way on AArch64 and is required?